orttraining/orttraining/core/graph/optimizer/lamb_optimizer_builder.cc

```diff
@@ -80,7 +103,16 @@ Status LambOptimizerBuilder::Build(
   // At the end of each Lamb call, the update count may be increased by one.
   const std::string step_tensor_name = "Step";  // per weight optimizer requires a per weight update count
   // Add step as an initializer.
-  new_external_initializers.emplace_back(CreateTensorProto<int64_t>(step_tensor_name, 1));
+  TensorProto step_tensor_proto;
+  const auto step_state_it = shared_optim_state.find(step_tensor_name);
+  if (step_state_it != shared_optim_state.end()) {
+    const auto& init_tensor = step_state_it->second.Get<Tensor>();
+    ORT_THROW_IF_ERROR(IsMatchingTypeAndShape(init_tensor, ONNX_NAMESPACE::TensorProto_DataType_INT64, {1}));
```

Comment:
My understanding was that the optimizer states were returned in this format:
```
model_weight_name:
{
    "Moment_1": moment1_tensor,
    "Moment_2": moment2_tensor,
    "Update_Count": update_tensor # if optimizer is adam, absent otherwise
},
"shared_optimizer_state": # if optimizer is shared, absent otherwise. So far, only lamb optimizer uses this.
{
    "step": step_tensor # int array of size 1
}
```

Considering that the "step" is an array of size 1, would it be of type ```ONNX_NAMESPACE::TensorProto_DataType_INT64``` or would it be similar to the other types in this same map where elements would be ```ONNX_NAMESPACE::TensorProto_DataType_FLOAT32``` or ```ONNX_NAMESPACE::TensorProto_DataType_FLOAT16```?

In short, my question is does ```CreateGenericMLValue(nullptr, GetAllocator(), "", initializer.second, &ml_value, true);``` in the pybind file understand that a particular array is an int64 vs float32/float16 type?
Please let me know if my question was not clear. I wanted to make sure I understood this code well.

Reply:
It does handle numeric tensors, please take a look at the CreateTensor() method within onnxruntime/onnxruntime/python/onnxruntime_pybind_mlvalue.cc